pkg/asset/machines/alibabacloud/machines.go

```diff
@@ -104,16 +99,29 @@ func provider(clusterID string,
 		SystemDiskSize:     mpool.SystemDiskSize,
 		RegionID:           platform.Region,
 		ResourceGroupID:    platform.ResourceGroupID,
-		ZoneID:             az,
+		ZoneID:             zoneID,
 		UserDataSecret:     &corev1.LocalObjectReference{Name: userDataSecret},
 		CredentialsSecret:  &corev1.LocalObjectReference{Name: "alibabacloud-credentials"},
 		Tags:               tags,
 		InstanceName:       fmt.Sprintf("%s-%s", clusterID, role),
 		IoOptimized:        "optimized",
 		RAMRoleName:        fmt.Sprintf("%s-role-%s", clusterID, role),
 		SecurityGroups:     sgTags,
-		VSwitch:            &vSwitchTags,
-	}, nil
+	}
+
+	if vswitchID == "" {
+		vSwitchTags := alibabacloudprovider.ResourceTagReference{
+			Tags: append(tags, alibabacloudprovider.Tag{
```

Comment:
How does the alibaba machine-api provider work when trying to find the VSwitch? Do the tags on the VSwitch have to match exactly the tags specified in the machine spec? Or do the tags on the VSwitch just need to be a superset of the tags specified in the machine spec? I am a bit surprised that we are adding *all* of the tags for the machine to the tags that we are looking for on the VSwitch, as opposed to just including the "Name" tag.

Reply:
> Or do the tags on the VSwitch just need to be a superset of the tags specified in the machine spec?

Provider will match a superset of the tags specified in the machine spec.

>  I am a bit surprised that we are adding all of the tags for the machine to the tags that we are looking for on the VSwitch, as opposed to just including the "Name" tag.

In fact, the Name tag can be matched to the only VSwitch.